### PR TITLE
feat: add single_terminal_per_instance option for toggleterm

### LINKED ISSUE #209: Toggleterm executor opens new tab instead of rewriting existing one
My config is
```lua
return {
  "Civitasv/cmake-tools.nvim",
  config = function()
    require("cmake-tools").setup {
      cmake_executor = {
        name = "toggleterm",
        opts = {
          direction = "tab", -- 'vertical' | 'horizontal' | 'tab' | 'float'
          close_on_exit = false, -- whether close the terminal when exit
          auto_scroll = true, -- whether auto scroll to the bottom
        },
      },
    }
  end,
}
```
If i rerun `CMakeGenerate`, it creates new tab. I am not sure this is optimal behaviour. Can it just clear existing one and write cmake output into it. 

Following is more feature request, than bug, but it seeml logical to have one toggleterm output for each preset\variant\profile.

### PR DESCRIPTION
**Pull request by TheRealLorenz**
As the name implies, uses a single instance of toggleterm by closing the already opened one.

Closes #209

### CODE REVIEW

File: lua/cmake-tools/toggleterm.lua [hunk 1]
```diff
@@ -24,6 +24,9 @@ end
 
 function _toggleterm.run(cmd, env_script, env, args, cwd, opts, on_exit, on_output)
   _toggleterm.cmd = cmd .. " " .. table.concat(args, " ")
+  if opts.single_terminal_per_instance and _toggleterm.term then
+    _toggleterm.term:close()
+  end
```
Comment (Civitasv) [inline lua/cmake-tools/toggleterm.lua:29]: I think we can make this as default.

File: README.md [hunk 2]
```diff
@@ -114,6 +115,7 @@ require("cmake-tools").setup {
         direction = "float", -- 'vertical' | 'horizontal' | 'tab' | 'float'
         close_on_exit = false, -- whether close the terminal when exit
         auto_scroll = true, -- whether auto scroll to the bottom
+        single_terminal_per_instance = true, -- single instance, autocloses the opened one, if present
```
Comment (Civitasv) [inline README.md:118]: Maybe `singleton` is enough, what do you think?

Comment (TheRealLorenz) [reply README.md:118]: Sounds good to be, I was trying to be consistent with the terminal's settings

**Comment by Civitasv** [comment]
Thanks for your contrib!

**Pull request merged**